### PR TITLE
Music: update header button behavior

### PR DESCRIPTION
**Pull request by breville**
This work began with the goal of just disabling the "Start Over" button for read-only workspaces, but ended up doing a couple things:

- The refresh, undo, and redo buttons are no longer shown for read-only workspaces.
- The undo and redo buttons are properly disabled when not available, taking them out of keyboard navigation.

### CODE REVIEW

File: apps/src/music/views/undo-redo-buttons.module.scss [hunk 1]
```diff
@@ -22,5 +22,6 @@
   &Disabled {
     opacity: 50%;
     cursor: default;
+    pointer-events: none;
```
Comment (breville) [inline apps/src/music/views/undo-redo-buttons.module.scss]: This doesn't actually prevent selection by pressing Enter when using keyboard navigation, hence the check in code as well, but felt prudent.

File: apps/src/music/views/HeaderButtons.tsx [hunk 2]
```diff
@@ -73,7 +79,10 @@ const HeaderButtons: React.FunctionComponent<HeaderButtonsProps> = ({
       <button
```
Comment (bencodeorg) [inline apps/src/music/views/HeaderButtons.tsx]: Why not actually disable the button via the `disabled` attribute?

Comment (sanchitmalhotra126) [inline apps/src/music/views/HeaderButtons.tsx]: Should we just not render the button at all in this case?

Comment (breville) [reply apps/src/music/views/HeaderButtons.tsx]: Good call.  Will use this for the undo and redo buttons.

Comment (breville) [reply apps/src/music/views/HeaderButtons.tsx]: Good call.  Will apply this to three of the button: refresh, undo, redo.

**Comment by breville** [comment]
Thanks for the feedback, @bencodeorg and @sanchitmalhotra126.  I used both of your suggestions broadly.  I've ended up just hiding three of the buttons (refresh, undo, and redo) when the workspace is read-only.  And by applying the `disabled` attribute to the undo and redo buttons when they aren't enabled, they get taken out of keyboard navigation which seems the right approach.

File: apps/src/music/views/undo-redo-buttons.module.scss (hunk 1 above)
Comment (breville) [reply apps/src/music/views/undo-redo-buttons.module.scss]: Ended up doing a slightly different approach.

**Review by sanchitmalhotra126 — APPROVED** [review]

**Pull request merged**